apps/web/src/app/next_api/fetch-hsl-stops/route.ts

```diff
@@ -0,0 +1,229 @@
+import type {
+  ApolloQueryResult,
+  DefaultOptions,
+  DocumentNode,
+} from "@apollo/client";
+import { ApolloClient, gql, InMemoryCache } from "@apollo/client";
+import type {
+  Arrival,
+  ArrivalAttribute,
+  Data,
+  RenderableStop,
+  Stop,
+  StopFromApi,
+  StopOutData,
+  StopTime,
+  StopType,
+} from "../../../lib/types/hsl-helper-types.ts";
+
+const STOPS = [
+  // Metro east and west
+  ["HSL:2222603", "HSL:2222604"],
+  // Raide jokeri east and west
+  ["HSL:2222406", "HSL:2222405"],
+  // Aalto Yliopisto bus stop "east" and "west"
+  ["HSL:2222234", "HSL:2222212"],
+] as const;
+const N_ARRIVALS = 6;
+
+export const dynamic = "force-dynamic";
+
+export async function GET() {
+  const stops = await Promise.all(STOPS.map(getStop));
+
+  const dataFromHsl: RenderableStop[] = stops.filter(
+    <T>(stop: T | null): stop is T => stop !== null,
+  );
+
+  const retData = {
+    type: "Data",
+    data: dataFromHsl,
+  };
+
+  return Response.json(
+    { retData },
+    {
+      status: 200,
+      headers: {
+        "Cache-Control": "no-cache, no-store, must-revalidate",
+        Pragma: "no-cache, no-store",
+        Expires: "0",
+      },
+    },
+  );
+}
+
+const GetStopSchedule = (StopId: string): DocumentNode =>
+  gql(`
+  {
+    stop(id: "${StopId}") {
+      name
+        stoptimesWithoutPatterns {
+        realtimeArrival
+        realtimeArrival
+        serviceDay
+        headsign
+        trip{
+          routeShortName
+        }
+      }
+    }
+  }
+`);
+
+function pad(number: number, size: number) {
+  let s = String(number);
+  while (s.length < (size || 2)) {
+    s = "0".concat(s);
+  }
+  return s;
+}
+
+function removeSubstring(fullString: string): string {
+  const subStrings: string[] = [
+    " via Leppävaara",
+    " via Rautatientori",
+    " via Tapiola (M)",
+    " via Huopalahti as.",
+    " via Tapiola",
+    " via Pasila as.",
+  ];
+  let str = fullString;
+  for (const subString of subStrings) {
+    // HSL sometimes has a bug where HEadSign is null so this handles case string in is null :D
+    if (str) {
+      str = str.replace(subString, "");
+    } else {
+      ("Null");
```

Comment:
Is this line doing anything? It doesn't look like the string value is returned or used at all, could this else branch be removed completely?